tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/CachedBasicPropertiesHelper.cs

```diff
@@ -1,23 +1,44 @@
-﻿// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
+// <copyright file="CachedBasicPropertiesHelper.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
 #nullable enable
+using System;
+using System.Reflection.Emit;
+using System.Runtime.InteropServices;
+using Datadog.Trace.DuckTyping;
 using Datadog.Trace.Util;
 
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.RabbitMQ;
 
 internal static class CachedBasicPropertiesHelper<TBasicProperties>
 {
     // ReSharper disable once StaticMemberInGenericType
-    private static readonly ActivatorHelper Activator;
+    private static readonly Func<object, object> Activator;
 
     static CachedBasicPropertiesHelper()
     {
-        Activator = new ActivatorHelper(typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!);
+        var targetType = typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.BasicProperties")!;
+        var parameterType = typeof(TBasicProperties).Assembly.GetType("RabbitMQ.Client.IReadOnlyBasicProperties")!;
+
+        var constructor = targetType.GetConstructor([parameterType])!;
+
+        var createBasicPropertiesMethod = new DynamicMethod(
+            $"TypeActivator_{targetType.Name}_{parameterType.Name}",
+            targetType,
+            [typeof(object)],
+            typeof(DuckType).Module,
+            true);
+
+        var il = createBasicPropertiesMethod.GetILGenerator();
+        il.Emit(OpCodes.Ldarg_0); // Load first argument (object).
+        il.Emit(OpCodes.Castclass, parameterType); // Cast to IReadOnlyBasicProperties
+        il.Emit(OpCodes.Newobj, constructor);
+        il.Emit(OpCodes.Ret);
+        Activator = (Func<object, object>)createBasicPropertiesMethod.CreateDelegate(typeof(Func<object, object>), targetType);
```

Comment:
@johang88: After some debugging, we figured out that the call to `CreateDelegate()` here was throwing an exception about the method signatures not matching. I fixed the issue by removing the `targetType` parameter, since the dynamic method will not be bound to a type (i.e. it acts like a static method, not an instance method).

We had to copy your commits into a new branch in the repository to run all our tests: see #6753. Once we get that new PR reviewed and approved, and all tests pass, we will merge that one in favor of this one. Your commits are still attributed to you as the author, though. Thanks for your help with this issue.